crates/ruff/src/rules/pylint/rules/eq_without_hash.rs

```diff
@@ -65,9 +66,31 @@ fn has_eq_without_hash(body: &[Stmt]) -> bool {
     let mut has_hash = false;
     let mut has_eq = false;
     for statement in body {
+        // check if `__hash__` was explicitly set to `None`
+        if let Stmt::Assign(ast::StmtAssign { targets, value, .. }) = statement {
+            let [target] = targets.as_slice() else {
+                continue;
+            };
+
+            let is_name_hash = target
+                .as_name_expr()
+                .is_some_and(|expr| expr.id == "__hash__");
+            has_hash = is_name_hash
+                && matches!(
+                    value.as_ref(),
+                    Expr::Constant(ast::ExprConstant {
+                        value: Constant::None,
```

Comment:
Does this need to be `None` or would assignment work?

Reply:
> While the purpose of PLW1641 is exactly to check for when `__eq__` is defined and `__hash__` is None (the default when `__eq__` is defined but `__hash__` is not), I think that if the code explicitly sets `__hash__ = None` then the lint should not trigger since the programmer explicitly asks for it.

It needs to be `None` according to the issue description.

Reply:
👍